### ISSUE TITLE
Setting mat3 row doesn't transpile correctly

### ISSUE THREAD

**Issue by DVLP** [stackgl/glsl-transpiler#56]
Declaration
`varying mat3 mat3one;`

With value assignment
`mat3one[0] = vec3(1., 1., 0.);`

Transpiles to
`mat3one.slice(0, 3) = [1., 1., 0.];`

Which doesn't work. 
With arrays of matrices 
`varying mat3 mat3Arr[15];`
`mat3Arr[4][0] = vec3(1., 1., 0.)` 
transpiles to 
`= [1, 1, 0]`

**Comment by dy** [comment]
yeah, that's tricky one. Not sure if it's better to manually detect all possible kinds of left-hand target or convert all arrays to Float32Array and use subarray+reduce, sort of 
`([1,1,0]).reduce((arr,v,i)=>(arr[i]=v,arr), mat3Arr.subarray(0,3))`

**Comment by dy** [comment]
Done as outlined case. Array of matrices is not yet supported, needs explicit effort.

**Comment by DVLP** [comment]
Nice! For now I only updated the package and it seems to be still working. I'll try some matrix read/writes tomorrow
